source/javascript/src/utils/Router.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Routing class used to help with navigating between pages
+ * for our site.
+ * @author William Chung <wchung@ucsd.edu>, Noah Teshima <nteshima@ucsd.edu>
+ */
+class Router {
+  /**
+     * Constructor used to create a URL from the current page location.
+     * @param {Object[]} params Rest parameter reference containing information
+     * for creating the relevant route. If no argument is provided, we initialize
+     * the corresponding URL with the href of the current page. Otherwise,
+     * we assume the first argument is a URL reference containing the
+     * URL to use for navigation.
+     */
+  constructor (...params) {
+    switch (params.length) {
+      case 0:
+        this._url = new URL(window.location.href)
+        break
+      case 1:
+        this._url = params[0]
+        break
+    }
+  }
+
+  /**
+     * Sets the URL to use for navigation or parameter parsing.
+     * @param {URL} url URL object containing the
+     * URL to use for navigation.
+     */
+  set url (url) {
+    this._url = url
+  }
+
+  /**
+     * @returns {URL} URL object containing the
+     * URL to use for navigation.
+     */
+  get url () {
+    return this._url
+  }
+
+  /**
+     * Subroutine of Router which changes the page's href
+     * to the value of the field url
+     */
+  navigate () {
+    window.location.href = this.toString()
```

Comment:
sometimes the parameter could be encoded. For example, from my experience with collection, if there is a space in the collection name, it will be encoded as /%20/g. so we might have to watch out for this

Reply:
Didn't realize this before; thanks for brining it up. We can address it specifically as a test case for @rdoron  or @klbaker  to include before Tuesday.

Reply:
Sounds good. You can go ahead and merge this pr.